src/v/cluster/topic_table_probe.cc

```diff
@@ -9,14 +9,133 @@
 
 #include "cluster/topic_table_probe.h"
 
+#include "cluster/cluster_utils.h"
 #include "cluster/topic_table.h"
 #include "config/configuration.h"
+#include "config/node_config.h"
 #include "prometheus/prometheus_sanitize.h"
 
 namespace cluster {
 
 topic_table_probe::topic_table_probe(const topic_table& topic_table)
-  : _topic_table(topic_table) {}
+  : _topic_table(topic_table) {
+    setup_metrics();
+}
+
+void topic_table_probe::setup_metrics() {
+    setup_public_metrics();
+    setup_internal_metrics();
+}
+
+void topic_table_probe::setup_internal_metrics() {
+    if (
+      config::shard_local_cfg().disable_metrics() || ss::this_shard_id() != 0) {
+        return;
+    }
+
+    // TODO: Delete this metric after public is supported in scraping
+    // configurations
+    namespace sm = ss::metrics;
+    _internal_metrics.add_group(
+      prometheus_sanitize::metrics_name("cluster:partition"),
+      {
+        sm::make_gauge(
+          "moving_to_node",
+          [this] { return moving_to_partitions; },
+          sm::description("Amount of partitions that are moving to node"))
+          .aggregate({sm::shard_label}),
+        sm::make_gauge(
+          "moving_from_node",
+          [this] { return moving_from_partitions; },
+          sm::description("Amount of partitions that are moving from node"))
+          .aggregate({sm::shard_label}),
+        sm::make_gauge(
+          "node_cancelled_movements",
+          [this] { return cancelled_movements; },
+          sm::description("Amount of cancelled partition movements for node"))
```

Comment:
Maybe call it not cancelled but cancelling (`cancelling_movements_on_node`). The word "cancelled" hints that it is a cumulative count